### PR TITLE
Add notes to group summaries for components that are not installed by default

### PR DESCRIPTION
**Pull request by allclark**

### CODE REVIEW

**Comment by allclark** [comment]
#sign-off

**Comment by tjprescott** [comment]
Hi @allclark there's a few issues here. The first is that unless a package is installed, you won't see it when you run `az -h`. If you run `az {component} -h` for something that isn't installed, you'll just get a usage error. So the long summaries will never show unless you already have the component installed. The other problem is that the packaged releases (apt-get, brew, etc.) don't support the `az component` command and so you can't install those components.

**Comment by allclark** [comment]
@tjpresscott - the comment is for people who find the online docs. We've had customer feedback that this happened: "I've just installed AZ and cannot use keyvault for shatever reason.   I suspect I don't have the component installed, but it would be nice to have something here to tell me how to do so." I will reword these, though.

**Comment by allclark** [comment]
@tjprescott How do packaged releases (apt-get, etc), get components like keyvaultl and iot?

**Comment by tjprescott** [comment]
@allclark the current plan is that they don't. If they want to install the non-default components they would need to do one of the installs that permits this model (pip, curl). KeyVault will likely be promoted to the default package list soon.

cc/ @derekbekoe

**Comment by allclark** [comment]
@tjprescott OK. That sounds like it needs to be explained, too. I'll take a crack at it. I assume pip or curl is supported on all of the platforms where we're suggesting a packaged release. Is that right? cc @derekbekoe

**Comment by derekbekoe** [comment]
>  I assume pip or curl is supported on all of the platforms where we're suggesting a packaged release. Is that right?

Yes that's correct.

File: src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_help.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 helps['iot'] = """
     type: group
     short-summary: Commands to connect, monitor, and control millions of IoT assets
+    long-summary: If you don't have the keyvault component installed, add it with `az component update --add iot`
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_help.py]: keyvault -> iot as this is the iot component.

**Pull request merged**